component/core/src/main/java/org/exoplatform/social/core/search/impl/SearchServiceImpl.java

```diff
@@ -76,6 +80,9 @@ public void addConnector(SearchConnectorPlugin connectorPlugin) {
 
   @Override
   public Set<SearchConnector> getEnabledConnectors() {
+    UserACL userACL = CommonsUtils.getService(UserACL.class);
+    boolean isExternalUser = userACL.isUserInGroup(PLATFORM_EXTERNALS_GROUP);
+    connectors.stream().filter(connector -> connector.getName().equals("space") || connector.getName().equals("people")).forEach(connector -> connector.setEnabled(!isExternalUser));
```

Comment:
Can you ensure this : 
1) an external user goes to unified search : he sees all connectors excepts people and space
2) log with another user, which is internal. Does he see the connectors spaces and people ? 

As developed, I'm afraid that the setEnabled can disabled connectors for all users.

Reply:
Yes it's , since it is not stored in the **settingService** and this method is called each time the search portlet is instantiated.

Reply:
But you modify the list connectors at runtine, no ? So the next user which will call getEnabledConnectors will get a list with some connetors disabeld

Reply:
it depends on user status (internal or external) `connector.setEnabled(!isExternalUser)`

Reply:
In fact, the connectors list is an object present in SearchServiceImpl. So if you call setEnabled(false) for one connector of this list, you will set it disable for all users, and this is KO.